packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,185 @@
+   */
+  LINEAR = 'Linear'
+}
+
+/**
+ * Properties of a reference to a CodeDeploy Lambda Deployment Configuration.
+ */
+export interface CustomLambdaDeploymentConfigProps {
+
+  /**
+   * The type of deployment config, either CANARY or LINEAR
+   */
+  readonly type: CustomLambdaDeploymentConfigType;
+
+  /**
+   * The integer percentage of traffic to shift:
+   * - For LINEAR, the percentage to shift every interval
+   * - For CANARY, the percentage to shift until the interval passes, before the full deployment
+   */
+  readonly percentage: number;
+
+  /**
+   * The interval, in number of minutes:
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: Duration;
+
+  /**
+   * The verbatim name of the deployment config. Must be unique per account/region.
+   * Other parameters cannot be updated if this name is provided.
+   * @default - automatically generated name
+   */
+  readonly deploymentConfigName?: string;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ * @resource AWS::CodeDeploy::DeploymentGroup
+ */
+export class CustomLambdaDeploymentConfig extends Resource implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    this.validateParameters(props);
+
+    // In this section we make the argument for the AWS API call
+    // The name of the config is <construct unique id>.Lambda<deployment type><percentage>Percent<interval>Minutes
+    // Unless the user provides an explicit name
+
+    const deploymentType = 'TimeBased' + props.type.toString();
+    const intervalMinutes = props.interval.toMinutes().toString();
+    const percentage = props.percentage.toString();
+
+    let routingConfig; // The argument to the AWS API call
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: intervalMinutes,
+          canaryPercentage: percentage,
+        },
+      };
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: intervalMinutes,
+          linearPercentage: percentage,
+        },
+      };
+    }
+
+    // What the physical resource ID is for Cloudformation
+    // Designed so that, if the construct name, percentage, or interval changes,
+    // the deployment config will be deleted and a new one created in its place
+    const resourceName = this.generatePhysicalResourceId(props);
```

Comment:
I appreciate your thoughtfulness here, but this is still needlessly complicated. I don't want to wade through this code to find out that actually `generatePhysicalResourceId` and `generateDeploymentConfigName` are the same thing.

Please, get rid of `resourceName`, and just use `this.deploymentConfigName` everywhere. I think that will make the code much clearer.